README.md

```diff
@@ -1,48 +1,75 @@
-#####   4) URL of your stack (if the endpoint is exposed)
-
-### Q. What if i successfully complete all the challenges ?
-#####  We have many more for you to solve as a member of the Rearc team !
-
-### Q. What if i find a bug ?
-#####  Awesome! Lets talk.
-
-### Q. What if i fail ?
-#####  Do. Or do not. There is no fail.
-
-### Q. Can i share this quest with others ?
-##### No.
+### Q. What is this quest?
+
+It is a fun way to assess your cloud skills. It is also a good representative sample of the work we do at Rearc. Quest is a webapp made with node.js and golang.
+
+### Q. Do I need to be an expert in node.js and golang?
+
+No. The starting point of the quest app is `npm install && npm start`. That is all the node.js you need to know. And you wont even see golang.
+
+### Q. So what skills should I have?
+
+AWS. General concepts, especially networking. Docker (containerization). IaC (Infrastructure as code). Linux/Unix. Git. TLS certs is a plus.
+
+### Q. What do I have to do?
+
+1. If you know how to use git, start a git repository (local-only is acceptable) and commit all of your work to it.
+2. Deploy the app in AWS and navigate to the index page. Use Linux 64-bit as your OS (Amazon Linux preferred).
+3. Deploy the app in a Docker container. Use `node:10` as the base image.
+4. Inject an environment variable (`SECRET_WORD`) in the Docker container. The value of `SECRET_WORD` should be the secret word discovered on the index page of the application.
+5. Deploy a load balancer in front of the app.
+6. Complete Terraform-ing and/or CloudFormation-ing the entire stack for "single click" deployment (Please use Terraform v1+).
+7. Add TLS (https). You may use locally-generated certs.
+
+### Q. How do I know I have solved these stages?
+
+Each stage can be tested as follows (where <ip_or_host> is the location where the app is deployed) 
+
+1. Index page (contains the secret word) - `http(s)://<ip_or_host>[:port]/`
+2. Docker check - `http(s)://<ip_or_host>[:port]/docker`
+3. Secret Word check - `http(s)://<ip_or_host>[:port]/secret_word`
+4. Load Balancer check  - `http(s)://<ip_or_host>[:port]/loadbalanced`
+5. TLS check - `http(s)://<ip_or_host>[:port]/tls`
+6. Terraform and/or CloudFormation - Manually checked when your quest is submitted.
+
+### Q. Do I have to do all these?
+
+You may do whichever, and however many, of the tasks above as you'd like. We suspect that once you start, you won't be able to stop. It's addictive.
+
+### Q. What do I have to submit?
+
+1. Your work assets, as one or both of the following:
+  - A link to a hosted git repository.
+  - A ZIP file containing your project directory. Include the `.git` sub-directory if you used git.
+2. Proof of completion, as one or both of the following:
+  - A link to a hosted AWS deployment.
+  - One or more screenshots showing, at least, the index page of the final deployment.
+
+Your work assets should include:
+
+- Terraform or CloudFormation modules, if you completed that task.
```

Comment:
I suggest we add AWS CDK to the list here as well.

Reply:
Added CDK.

Side note: we might want to eventually think about letting people host this on at least Google Cloud as well, if they choose to do so, as it's still relatively similar to AWS. I wouldn't mind allowing Azure deployments either but it does differ significantly from AWS and GCP.